Merge 2022-05-01-GA authoring updates and "LUIS" string case update into 2022-05-15-preview branch

bidisha-c opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

bidisha-c committed:
Merge 2022-05-01-GA files changes into 2022-05-15-preview branch

heaths commented:
Take a look at the SpellCheck and Avocado errors. Both should be easy to fix. The AVocado errors look like they are caused by the fact there are different files from the readme.md changes you seemed to have copied.

heaths commented:
Actually, looks like you didn't update readme.md to accommodate the two -authoring.json files.

heaths commented:
I'm not able to generate a semantic diff using OpenAPI Hub: https://portal.azure-devex-tools.com/tools/diff/eyJzb3VyY2VVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvcmVsZWFzZS1jb2duaXRpdmVzZXJ2aWNlcy1MYW5ndWFnZS0yMDIyLTA1LTAxLUdBL3NwZWNpZmljYXRpb24vY29nbml0aXZlc2VydmljZXMvZGF0YS1wbGFuZS9MYW5ndWFnZS9yZWFkbWUubWQiLCJ0YXJnZXRVcmwiOiJodHRwczovL3Jhdy5naXRodWJ1c2VyY29udGVudC5jb20vQXp1cmUvYXp1cmUtcmVzdC1hcGktc3BlY3MvOGY4YTU3MDY5MWMxMTY4YmUyZGUwMmNmNzhjMjU5ODJjMmRjYzM5Zi9zcGVjaWZpY2F0aW9uL2NvZ25pdGl2ZXNlcnZpY2VzL2RhdGEtcGxhbmUvTGFuZ3VhZ2UvcmVhZG1lLm1kIiwic291cmNlVGFnIjoicmVsZWFzZV8yMDIyXzA1XzAxIiwidGFyZ2V0VGFnIjoicmVsZWFzZV8yMDIyXzA1XzE1X3ByZXZpZXcifQ==

@peytonfraser this is an example of a critical process we'll almost never have to do again with the new process proposal.

bidisha-c commented:
Updating the readme file  and the custom words.

heaths reviewed (changes requested):
Since OpenAPI Hub wasn't working to semantically diff the swaggers from this merge branch to the 2022-05-01-GA branch, I did (using just git and the [GitHub CLI](https://github.com/cli/cli)):

```bash
git fetch upstream release-cognitiveservices-Language-2022-05-01-GA
gh pr checkout 18943
pushd specification\cognitiveservices\data-plane\Language
git checkout upstream/release-cognitiveservices-Language-2022-05-01-GA stable/2022-05-01
```

Using VSCode, I can then compare files and found the following comparing 2022-05-01 in the release branch to files in your PR merge branch:

1. ✅ Only change in _analyzeconversations-authoring.json_ is the `api-version`.
2. ✅ Apart from the `api-version` change in _analyzeconersations.json_, I see:
   1. ✅  `/analyzeconversations-jobs` endpoints added.
   2. ✅ `role` property added to `ConversationItemBase`
   3. ✅ Bunch of new models related to the new endpoints.
3. ✅ Only `api-version` change in _analyzetext-authoring.json_.
4. ✅ Apart from `api-version` change in _common.json_, I see: 
   1. ⚠️ Dropped `Timeout`, `QuotaExceeded`, `Conflict`, and `Warning` errors.
   2. ✅ Added models for `TaskState`, `JobState`, and related.
   3. ❔ New `InputError` and `InputWarning`. I thought we were going to use the standard `Error` model for both of these?
5. Apart from `api-version` change in _textanalytics.json_, I see:
   1. ✅ Bunch of changes to use definitions in _common.json_.
   2. ⚠️ Some properties made required. This will trigger breaking change warnings for subsequent releases after 2022-05-01 is merged to `main`. Can these be made required for GA now e.g., `PiiTaskResult.results`?
   3. ✅ New summary and PII models.

Marking off as "Request changes" mainly to make sure this isn't merged before discussing the concerns above. Mainly, it's the future-breaking-changes issues I noted that have me concerned, but the rest look good.

I didn't check over all the samples, but make sure the api-version is correct and that all samples are warranted i.e., that only samples for swaggers you're updating in 2022-05-15-preview are in there.

bidisha-c commented on specification/cognitiveservices/data-plane/Language/preview/2022-05-15-preview/analyzeconversations.json:
```diff
@@ -387,7 +387,7 @@
           "description": "The type of a target service.",
           "enum": [
-            "Luis",
+            "LUIS",
```
There is actually a PR that @ChongTang is working for to revert this. https://github.com/Azure/azure-rest-api-specs/pull/18840 I will also update my PR to reflect the same.

bidisha-c committed:
Address Avocado issues and revert case update for Luis

bidisha-c committed:
Add the missing error codes

bidisha-c committed:
Fix avocado

bidisha-c commented:
@heaths I have addressed the avocado and Spellcheck errors. Here are the other items you had questions/concerns about:
- Some properties made required. 
I have added a PR to add the parameters made "required" in this PR to the GA version. 

- Dropped Timeout, QuotaExceeded, Conflict, and Warning errors.
I have added the missing error codes in the common.json

- New InputError and InputWarning. 
"InputError" and "InputWarning" are basically after renamed "DocumentError" and "DocumentWarning" fields in the TextAnalytics.json so both analyze-text and analyze-conversations can share that. They both include the shared "Error" object.

bidisha-c committed:
Avocado fix

heaths commented:
@JeffreyRichter @johanste could you approve the breaking change? This is the required `error` property issue. After the 2022-05-01 GAs, this specific issue should go away. 🎉

johanste commented:
The breaking change is benign (and only relevant for client libraries) - an output-only type is more constrained than it used to be. If a client library/language wants to avoid a possible breaking change, then it can be transformed away.
